core/trino-main/src/main/java/io/trino/execution/SqlStageExecution.java

```diff
@@ -126,26 +126,23 @@ public StageId getStageId()
         return stateMachine.getStageId();
     }
 
-    public synchronized boolean transitionToFinished()
+    public synchronized void finish()
```

Comment:
would be cool to have simple renames like this one as separate preparatory commits.

Reply:
It is a little bit  more than a simple rename. The main goal was to change the semantics of this operation from a simple state transitioning (from what I understand the assumption was that when the transitionToFinish is called all the tasks should already be finished, as transitioning a running task to the ABORTED state  would result in a stage failure) to an operation that actually finishes stage execution (transition the stage to FINISHED state and cancel all the tasks that are still running).